Code/PgSQL/rdkit/adapter.cpp

```diff
@@ -30,6 +30,17 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
+
+// PostgreSQL 14 on Windows uses a hack to redefine the stat struct
+// The hack assumes that sys/stat.h will be imported for the first
+// time by win32_port.h, which is not necessarily the case
+// So we need to set the stage for the hack or it will fail
+#ifdef _WIN32
```

Comment:
is this always true for WIN32 or only when you are using MSVC, in which case it probably should be:
```suggestion
#ifdef _MSC_VER
```